tests/xfstests/partition.pm

```diff
@@ -281,6 +289,38 @@ sub format_with_options {
     }
 }
 
+sub install_dependencies_nfs {
+    my @deps = qw(
+      nfs-client
+      nfs-kernel-server
+      nfs4-acl-tools
+    );
+    zypper_call('in ' . join(' ', @deps));
+}
+
+sub setup_nfs_server {
+    my $nfsversion = shift;
+    assert_script_run('mkdir -p /export/test /export/scratch /nfs/test /nfs/scratch && chown nobody:nogroup /export/test /export/scratch && echo \'/export/test *(rw,no_subtree_check,no_root_squash)\' >> /etc/exports && echo \'/export/scratch *(rw,no_subtree_check,no_root_squash,fsid=1)\' >> /etc/exports');
+
+    my $nfsgrace = get_var('NFS_GRACE_TIME', 15);
+    assert_script_run("echo 'options lockd nlm_grace_period=$nfsgrace' >> /etc/modprobe.d/lockd.conf && echo 'options lockd nlm_timeout=5' >> /etc/modprobe.d/lockd.conf");
+
+    if ($nfsversion == '3') {
+        assert_script_run("echo 'MOUNT_NFS_V3=\"yes\"' >> /etc/sysconfig/nfs");
+        assert_script_run("echo 'MOUNT_NFS_DEFAULT_PROTOCOL=3' >> /etc/sysconfig/autofs && echo 'OPTIONS=\"-O vers=3\"' >> /etc/sysconfig/autofs");
+        assert_script_run("echo 'Defaultvers=3' >> /etc/nfsmount.conf && echo 'Nfsvers=3' >> /etc/nfsmount.conf");
+    }
+    else {
+        assert_script_run("sed -i 's/NFSV4LEASETIME=\"\"/NFSV4LEASETIME=\"$nfsgrace\"/' /etc/sysconfig/nfs");
+        assert_script_run("echo -e '[nfsd]\\ngrace-time=$nfsgrace\\nlease-time=$nfsgrace' > /etc/nfs.conf.local");
+    }
+    assert_script_run('exportfs -a && systemctl restart rpcbind && systemctl enable nfs-server.service && systemctl restart nfs-server');
+
+    # There's a graceful time we need to wait before using the NFS server
+    my $gracetime = script_output('cat /proc/fs/nfsd/nfsv4gracetime;');
+    sleep($gracetime * 2);
+}
+
```

Comment:
The code is *almost* the same as the one in `tests/xfstests/partition.pm`. I wonder if it made sense to create `lib/xfstests.pm` and put function there.

Reply:
> The code is _almost_ the same as the one in `tests/xfstests/partition.pm`. I wonder if it made sense to create `lib/xfstests.pm` and put function there.

The code you look into is just what we use for xfstests, no worry about duplicate :D
But if further tests still could use this code, I'll consider reorganizing it to lib folder in that moment. Thank you for the suggestion!